### PR TITLE
Plane Autoland

### PR DESCRIPTION
**Pull request by Rxann**
This feature fully and automatically lands any plane at LSIA or Fort Zancudo

### CODE REVIEW

**Comment by gir489returns** [comment]
Why is this a PR and not a Lua script? This seems fucking stupid, what use of this is to someone hacking?

**Comment by Rxann** [comment]
> Why is this a PR and not a Lua script? This seems fucking stupid, what use of this is to someone hacking?

Because just because you are fucking hacking doesn't mean you don't wanna have some fun.

**Comment by gir489returns** [comment]
You didn't answer my question. Why is this a PR and not a Lua file. This is garbage.

**Comment by Rxann** [comment]
It's not a Lua script because I don't care to make it one. And how the hell is it garbage if it works perfectly and doesn't add any downsides to the menu?

**Comment by gir489returns** [comment]
Because you suck at coding that's why. All your code is god awful, you don't know anything about reverse engineering, and you keep making PRs that are just fucking substandard. Do you even fucking use git dig before committing? Holy shit, you commit the dumbest things. This should be a Lua file, I vote to reject this from the repo.

**Comment by Rxann** [comment]
> Because you suck at coding that's why. All your code is god awful, you don't know anything about reverse engineering, and you keep making PRs that are just fucking substandard. Do you even fucking use git dif before committing? Holy shit, you commit the dumbest things. This should be a Lua file, I vote to reject this from the repo.

How the fuck is this code god awful. Secondly, I didn't know I needed to know reverse engineering to contribute a couple fucking features to this menu. And maybe I fucked the git diff but how the hell does that effect you. The code is the fucking same in the end. And just because there is a fucking lua api doesn't mean that every single little feature needs to be implemented. I think you need to get out of your fucking chair and go outside because you are so damn miserable all the time hating on me. And if my code is so damn bad at least you could fucking help me or give me some pointers on what I could do to better it like a nice person.

**Comment by Rxann** [comment]
And it's also a little feature, so it's not like there are any downsides from adding it.

**Comment by holysnipz** [comment]
move to `view_fun_vehicle` maybe? idk

**Comment by Rxann** [comment]
I mean I personally think it's fine but If other people think I need to I will

**Comment by gir489returns** [comment]
> How the fuck is this code god awful. Secondly, I didn't know I needed to know reverse engineering to contribute a couple fucking features to this menu. And maybe I fucked the git diff but how the hell does that effect you. The code is the fucking same in the end. And just because there is a fucking lua api doesn't mean that every single little feature needs to be implemented. I think you need to get out of your fucking chair and go outside because you are so damn miserable all the time hating on me. And if my code is so damn bad at least you could fucking help me or give me some pointers on what I could do to better it like a nice person.

No, because all you keep fucking doing is pushing random garbage shit to the repo without doing any critical thinking, and none of it is useful or applicable to the project goals of the base, you're just trying to fit in and like to yourself you are a mod developer or some bullshit like that. You're not a programmer, you're not an integrator, you're not a reverse engineer, you're not a designer and you're certainly not a fucking expert on any related field. So what the fuck are you? Useless, that's what you are. I've given you multiple feedbacks on PRs before and showed exactly what you were doing wrong and how to fix it, yet every single fucking new commit you make has the same mistakes, it's like screaming into the fucking void with you, and I'm just checked out, and now little Timmy made a fucking autopilot commit that could be a fucking 10 line Lua script and thinks he's super proud and accomplished now, go fucking suck eggs, dude.

**Comment by zilinwsx** [comment]
6

File: src/views/vehicle/view_vehicle.cpp [hunk 1]
```diff
@@ -184,6 +184,37 @@ namespace big
 			ImGui::SameLine();
 			ImGui::RadioButton(speed_unit_strings[(int)SpeedUnit::MPS].c_str(), (int*)&g.vehicle.speed_unit, (int)SpeedUnit::MPS);
 		}
+		ImGui::SeparatorText("Autolanding");
+		{
+			if (ImGui::Button("AutoLand at LSIA"))
```
Comment (tupoy-ya) [inline src/views/vehicle/view_vehicle.cpp]: You should use `components::button` it has the fiber_pool wrapper inside.
```cpp
components::button("AutoLand at LSIA", [] {
	if (VEHICLE::IS_THIS_MODEL_A_PLANE(ENTITY::GET_ENTITY_MODEL(self::veh)))
	{
		TASK::TASK_PLANE_LAND(self::ped, self::veh, -1024.981201, -3128.771729, 13.944453, -1370.038574, -2929.581543, 13.944449);
		g_notification_service->push_success("Autoland", "Starting Autoland at LSIA...");
	}
}
```

File: src/views/vehicle/view_vehicle.cpp [hunk 2]
```diff
@@ -184,6 +184,37 @@ namespace big
...
+			{
+				g_fiber_pool->queue_job([] {
+					if (VEHICLE::IS_THIS_MODEL_A_PLANE(ENTITY::GET_ENTITY_MODEL(self::veh)))
+					{
+						TASK::TASK_PLANE_LAND(self::ped, self::veh, -2041.194458, 2877.726074, 32.925156, -2141.520264, 2935.591309, 32.890343);
+						g_notification_service->push_success("Autoland", "Starting Autoland at Fort Zancudo...");
+					}
+				});
+			}
+			if (ImGui::Button("Cancel Autoland"))
```
Comment (tupoy-ya) [inline src/views/vehicle/view_vehicle.cpp]: Just name it clear tasks.
Also does it kick you out of your vehicle?

**Review by tupoy-ya — CHANGES_REQUESTED** [review]
I'm personally against merging this, we already have too much things that should be a lua script.

**Comment by ghost** [comment]
> Because you suck at coding that's why. All your code is god awful, you don't know anything about reverse engineering, and you keep making PRs that are just fucking substandard. Do you even fucking use git dif before committing? Holy shit, you commit the dumbest things. This should be a Lua file, I vote to reject this from the repo.

gamers, i have come to say the n word

**Comment by Rxann** [comment]
> > How the fuck is this code god awful. Secondly, I didn't know I needed to know reverse engineering to contribute a couple fucking features to this menu. And maybe I fucked the git diff but how the hell does that effect you. The code is the fucking same in the end. And just because there is a fucking lua api doesn't mean that every single little feature needs to be implemented. I think you need to get out of your fucking chair and go outside because you are so damn miserable all the time hating on me. And if my code is so damn bad at least you could fucking help me or give me some pointers on what I could do to better it like a nice person.
> 
> 
> 
> No, because all you keep fucking doing is pushing random garbage shit to the repo without doing any critical thinking, and none of it is useful or applicable to the project goals of the base, you're just trying to fit in and lie to yourself you are a mod developer or some bullshit like that. You're not a programmer, you're not an integrator, you're not a reverse engineer, you're not a designer and you're certainly not a fucking expert on any related field. So, what the fuck are you? Useless, that's what you are. I've given you multiple feedbacks on PRs before and showed exactly what you were doing wrong and how to fix it, yet every single fucking new commit you make has the same mistakes, it's like screaming into the fucking void with you, and I'm just checked out, and now little Timmy made a fucking autopilot commit that could be a fucking 10 line Lua script and thinks he's super proud and accomplished now, go fucking suck eggs, dude.


I don't know if this can fit through your small fucking brain but you have to do some fucking thinking to even code something. And despite what you fucking think and I don't know if you figured this out but your a programmer if you program something. And on every single fucking Pr you commented on I answered your fucking advice and tried to incorporate it. And also I don't see anywhere in a fucking file that says I can' contribute small features. So fuck off and maybe go spend some time outdoors because I can tell your fucking miserable all the time.

File: src/views/vehicle/view_vehicle.cpp (hunk 2 above)
Comment (Rxann) [reply src/views/vehicle/view_vehicle.cpp]: No, it does not

**Comment by Rxann** [comment]
> I'm personally against merging this, we already have too much things that should be a lua script.

I feel that that Lua scripting for the menu isn't quite at a point yet for little features like this to be turned into one, but if that's what you guys want me to do and think is best

**Comment by xiaoxiao921** [comment]
I just checked and this whole pr can be turned into a Lua script though

**Comment by Rxann** [comment]
If I turn it into a Lua script where should I upload it too? is there a marketplace of sorts for lua scripts?

**Comment by Yimura** [comment]
> If I turn it into a Lua script where should I upload it too? is there a marketplace of sorts for lua scripts?

You can submit a request on the [YimMenu-Lua/submission](https://github.com/YimMenu-Lua/submission) repository by creating an issue and filling in the form.

[Direct Link to Submit a Request](https://github.com/YimMenu-Lua/submission/issues/new/choose)

**Comment by Rxann** [comment]
> > If I turn it into a Lua script where should I upload it too? is there a marketplace of sorts for lua scripts?
> 
> 
> 
> You can submit a request on the [YimMenu-Lua/submission](https://github.com/YimMenu-Lua/submission) repository by creating an issue and filling in the form.
> 
> 
> 
> [Direct Link to Submit a Request](https://github.com/YimMenu-Lua/submission/issues/new/choose)

Ok, thanks for this. By the way yim.gta.menu/lua is giving a 404

**Comment by Yimura** [comment]
Closing this in favour for a Lua script, you can combine this and that other PR you made into one Lua script that you use to add features to YimMenu.

**Pull request closed without merging**